src/orange/detail/UnitInserter.cc

```diff
@@ -373,6 +382,38 @@ VolumeRecord UnitInserter::insert_volume(SurfacesRecord const& surf_record,
     return output;
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Process a single oriented bounding zone record.
+ */
+void UnitInserter::process_obz_record(VolumeRecord* vol_record,
+                                      OrientedBoundingZoneInput const& obz_input)
+{
+    CELER_EXPECT(obz_input);
+
+    OrientedBoundingZoneRecord obz_record;
+
+    // Set half widths
+    auto inner_hw_id = fast_real3s_.push_back(
+        calc_half_widths(calc_bumped_(obz_input.inner)));
+    auto outer_hw_id = fast_real3s_.push_back(
+        calc_half_widths(calc_bumped_(obz_input.outer)));
```

Comment:
There's no point in having an extra indirection into a separate vector of Real3 if you're not deduplicating or saving space by optimizing for the "no bbox" case with a false ID. Just replace `Array<Real3Id, 2> hw_ids;` in the OBZ record with `Array<Real3, 2> hw_ids;` and you'll save 2 ints per record and reduce memory accesses by avoiding the extra indirection.